### ISSUE TITLE
Completing a Neighborhood Bug

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#419]
I finally completed the Hillcrest neighborhood to test any lingering neighborhood completion bugs (https://github.com/ProjectSidewalk/SidewalkWebpage/issues/296). 

Unfortunately, I observed a few bugs (which seem to be related to calculating stats based on my old neighborhood, Hillcrest, vs. my newly assigned neighborhood, Randle Highlands). See: https://youtu.be/EQlLqLZ-aUg?t=16m4s (watch in HD).

First, right when I finish auditing Hillcrest, the background window is in an inconsistent state. It says that the Current Neighborhood is Randle Highlands, DC and to 'Audit 1000ft of this neighborhood' but the mission bar is 97% complete. Also, miles walked is 1.5 miles and labels is 2965--not sure where this came from. Circled in red below.

![image](https://cloud.githubusercontent.com/assets/1621749/20322360/bbfa5664-ab46-11e6-9a5a-6f288878c150.png)

Second, the Mission Complete dialog is messed up. It has the wrong label count, negative miles audited, etc.
![image](https://cloud.githubusercontent.com/assets/1621749/20322394/e00d9200-ab46-11e6-9bbc-09a53a15b43c.png)

Third, the map interface centered over Randle Highlands rather than my just completed Hillcrest neighborhood:
![image](https://cloud.githubusercontent.com/assets/1621749/20322426/0afa5cd2-ab47-11e6-8c4a-dbae5cb605c1.png)

**Comment by jonfroehlich** [comment]
Note that I never started a mission in Randle Highlands in the above thread; however, I did just now. Here's what happened.

First, the start mission screen is weird. I'm at 94% with 1.5 miles audited and 261 labels applied. Is this accurate? It might be simply because some of the roads I audited in Hillcrest went in to Randle Highlands--so I'm probably getting credit for that. Still, it's confusing. :)
![image](https://cloud.githubusercontent.com/assets/1621749/20345042/6bfdb1b0-abc3-11e6-8e55-1a68c75adb8e.png)

Second, when I finished the very short mission (only had 6% to go remember), I was not actually greeted with any green roads on the Mission Complete screen. All looked gray.
![image](https://cloud.githubusercontent.com/assets/1621749/20345177/f5b57b54-abc3-11e6-9a6e-82d9a695c54a.png)

**Comment by Soben713** [comment]
**An update on this issue:**

I dumped the database near the end of a neighborhood and kept testing things that happen when that neighborhood is finished and I encountered some bugs, in addition to what Jon mentioned in this thread. To fix all these issues I came up with another mechanism of what should happen when neighborhoods are completely audited that makes the code much simpler. We can discuss it in a quick project meeting.

I am not submitting a pull request for now because I just completed another neighborhood and surprisingly found another bug which didn't happen in the previous neighborhood. I need a break for now, but I'll continue working on it tomorrow and I'll submit the pull-request when I make sure there are no more bugs.
